features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.drawable.Drawable
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+@Suppress("MagicNumber")
+internal object Base64LRUCache : Cache, ComponentCallbacks2 {
+    @Suppress("UnsafeThirdPartyFunctionCall") // maxMemory does not throw
+    private val CACHE_MEMORY_SIZE = (Runtime.getRuntime().maxMemory() / 8).toInt()
+    private val ON_LOW_MEMORY_SIZE = CACHE_MEMORY_SIZE / 2 // 50% size
+    private val ON_MODERATE_MEMORY_SIZE = (CACHE_MEMORY_SIZE / 4) * 3 // 75% size
+
+    private var cache: LruCache<Int, String> = object : LruCache<Int, String>(CACHE_MEMORY_SIZE) {
+        override fun sizeOf(key: Int?, value: String): Int {
+            return value.toByteArray().size
```

Comment:
`toByteArray()` will involve new byte array allocation each time (according to `String.getBytes`), so effectively increasing memory footprint of our cache 2x if this method is called frequently.

Can we optimize it somehow? Not have a solution in my head, just throwing in the idea for optimization.

Reply:
Yes, I understand the direction. Base64 takes up 6 of 8 bits, so 0.75 * the length of the string would give us the size without creating a bytearray

Reply:
I think it will be the simply the length of the string, because even if we take only only lower half of the byte space, the atomic unit is still the byte.

Moreover, internally JVM at least (not sure about ART though) stores Strings in UTF-16 format, so probably the native size of the object will be `2 * length`? ignoring the fields of the `String` object itself (which is just few bytes).

Reply:
Thanks - it is indeed string length due to utf-8. I'll make these changes and it should improve memory quite a lot